src/http-client/node-http2-client.ts

```diff
@@ -0,0 +1,169 @@
+import http2 from "http2";
+
+import { HTTPClient, HTTPRequest, HTTPResponse } from "./index";
+import { NetworkError } from "../errors";
+import { QueryRequest } from "../wire-protocol";
+
+const { HTTP2_HEADER_PATH, HTTP2_HEADER_METHOD, HTTP2_HEADER_STATUS } =
+  http2.constants;
+
+/**
+ * An implementation for {@link HTTPClient} that uses the node http package
+ */
+export class NodeHTTP2Client implements HTTPClient {
```

Comment:
P2 - expose a close method in the HTTPClient interface that allows requests to complete and then closes down the session.

Reply:
i'll cut a JIRA for this.